phoenixRest/views/agenda/__init__.py

```diff
@@ -36,29 +36,79 @@ def __getitem__(self, key):
         node.__name__ = key
         return node
 
+
+
 @view_config(context=AgendaViews, request_method='GET', renderer='json', permission='get')
 def get_agenda_entries(request):
     # Find all events and sort them by start time
     entries = request.db.query(AgendaEntry).filter(AgendaEntry.event == get_current_event(request)).order_by(AgendaEntry.time.asc()).all()
     return entries 
 
+
+
 @view_config(context=AgendaViews, request_method='PUT', renderer='json', permission='create')
-@validate(json_body={'time': int, 'title': str, 'description': str, 'event_uuid': str})
+@validate(json_body={'event_uuid': str, 'title': str, 'description': str, 'location': str, 'time': int, 'state_pinned': bool})
 def create_agenda_entry(context, request):
+    # Attempt to find current Event uuid
     event = request.db.query(Event).filter(Event.uuid == request.json_body['event_uuid']).first()
 
     if not event:
         request.response.status = 404
         return {
             "error": "Event not found"
         }
-    parsed_time = datetime.fromtimestamp(int(request.json_body['time']))
 
-    entry = AgendaEntry(title=request.json_body['title'],
-                        description=request.json_body['description'],
-                        time=parsed_time, 
-                        event=event)
+    entry = AgendaEntry(
+        event=event,
+        title=request.json_body['title'],
+        description=request.json_body['description'],
+        location=request.json_body['location'],
+        time=datetime.fromtimestamp(int(request.json_body['time'])), 
+        state_pinned=request.json_body['state_pinned'],
+        log_created_by_uuid=request.user.uuid
+    )
     request.db.add(entry)
     request.db.flush()
     return entry
 
+
+
+@view_config(context=AgendaViews, request_method='PATCH', renderer='json', permission='create')
+@validate(json_body={'uuid': str, 'event_uuid': str, 'time': int, 'title': str, 'description': str, 'location': str, 'state_pinned': bool, 'state_deviating_time_unknown': bool, 'state_cancelled': bool})
```

Comment:
We use the `instance` pattern for this. The second you are passing in the uuid of the thing you are modifying in the http body, you are doing it wrong.

Reply:
Also avoid `state_` prefixes. They provide no new information.